src/snowflake/snowpark/dataframe.py

```diff
@@ -3486,18 +3497,22 @@ def create_or_replace_temp_view(
         return self._do_create_or_replace_view(
             formatted_name,
             LocalTempView(),
+            comment=comment,
             _statement_params=create_or_update_statement_params_with_query_tag(
                 statement_params or self._statement_params,
                 self._session.query_tag,
                 SKIP_LEVELS_TWO,
             ),
         )
 
-    def _do_create_or_replace_view(self, view_name: str, view_type: ViewType, **kwargs):
+    def _do_create_or_replace_view(
+        self, view_name: str, view_type: ViewType, comment: Optional[str], **kwargs
+    ):
         validate_object_name(view_name)
         cmd = CreateViewCommand(
             view_name,
             view_type,
+            comment,
             self._plan,
```

Comment:
there is a way you can do without changing the api:

```suggestion
        validate_object_name(view_name)
        cmd = CreateViewCommand(
            view_name,
            view_type,
            kwargs.get("comment"),
            self._plan,
```